npm/private/npm_link_package.bzl

```diff
@@ -163,8 +174,36 @@ If unset, the package name of the src npm_link_package_store is used.
 If set, takes precendance over the package name in the src npm_link_package_store.
 """,
     ),
+    "allow_unresolved_symlinks": attr.bool(
+        mandatory = True,
+        doc = """Whether unresolved symlinks are enabled in the current build configuration.
+
+        These are enabled with the --experimental_allow_unresolved_symlinks flag.
+
+        Typical usage of this rule is via a macro which automatically sets this
+        attribute based on a `config_setting` rule.
+        """,
+    ),
 }
 
+def _make_symlink(ctx, symlink_path, target_file):
+    files = []
+    if ctx.attr.allow_unresolved_symlinks:
+        symlink = ctx.actions.declare_symlink(symlink_path)
+        ctx.actions.symlink(
+            output = symlink,
+            target_path = relative_file(target_file, symlink.path),
+        )
+        files.append(target_file)
```

Comment:
Interesting, I didn't notice previously (before the helper func) that the new `declare_symlink` also requires the target file. That makes sense, but why did the old one not require it? Because it was absolute instead of relative 🤔

Reply:
The new `declare_symlink` takes a target path so the file itself has to be explicitly added as an input. That why they call the feature unresolved/dangling symlinks since you can symlink to something that is not an input.

The old method takes a `target_file` which is automatically added to the inputs transitively via the symlink file